main.go

```diff
@@ -0,0 +1,196 @@
+package main
+
+import (
+	"bytes"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/tls"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/json"
+	"encoding/pem"
+	"flag"
+	"fmt"
+	"github.com/Yonas-net/windows-crash/lib/base"
+	"io/ioutil"
+	"log"
+	"math/big"
+	_ "os"
+	"time"
+)
+
+type Encoder struct {
+	JSONRPC string  `json:"jsonrpc"`
+	Method  string  `json:"method"`
+	Params  ConfigInfo `json:"params"`
+}
+
+type ConfigInfo map[string]string
+
+func main()  {
+	port := flag.Int("port", 0, "On which port the server starting to listen")
+	host := flag.String("host", " ", "Host Name or IP Address")
+	cakey := flag.String("cakey", " ", "Private Key Certificate Chain")
+	cacert := flag.String("cacert", " ", "Certificate Authority")
+	concurrency := flag.Int("concurrency", 0, "concurrency")
+	sleep := flag.Duration("sleep", 0, "sleep in seconds")
+	duration := flag.Int("duration", 0, "Amount of files to be synced to the specified zone")
```

Comment:
Re-read the task, especially what VALUES the program shall take via which -params and what the VALUES are for.

Are you sure about the descriptions, especially of -port and -duration?